### PR TITLE
Core/ts version task 96 non exist routes

### PR DESCRIPTION
**Pull request by franken5tein**
## What does this MR do?

Handle non exist routes

## Check list  

- [x] All Merge Request pipelines should be passed.
- [x] Make sure all the changes are necessary and correct.
- [x] Make sure no needed change missing (all necessary changes are made).
- [x] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [ ] All necessary unit tests are written.
- [x] Critical changes should be tested locally (or on staging).

### CODE REVIEW

**Comment by hypersolo** [comment]
noice :100: 
now we are responding with 200 status code when a route doesn't exist. which should be 404. by doing this it is then ready to merge! :cupid:

**Comment by franken5tein** [comment]
> noice 100 now we are responding with 200 status code when a route doesn't exist. which should be 404. by doing this it is then ready to merge! cupid

DONE!

**Review by WilliamHawkingStephen — APPROVED** [review]

File: src/presentation/http/controllers/not-found.controller.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { NextFunction, Request, Response } from "express";
+import { BaseController } from "./base.controller";
+
+export class NotFoundController extends BaseController {
+  public get(_req: Request, res: Response, next: NextFunction) {
+    res.status(404).send({ success: false, data: { message: "Not Found" } });
+    next();
```
Comment (GreatGaryCooper) [inline src/presentation/http/controllers/not-found.controller.ts:7]: No need to call next, unless it's for logging purposes.

**Review by GreatGaryCooper — APPROVED** [review]
Awesome work, thanks @franken5tein 🌹

File: src/presentation/http/controllers/not-found.controller.ts (hunk 1 above)
Comment (GreatGaryCooper) [reply src/presentation/http/controllers/not-found.controller.ts:7]: Ignore this comment, `next` must be called.

**Review by hypersolo — APPROVED** [review]

**Pull request merged**